packages/components/src/components/domainproperties/dataset/actions.ts

```diff
@@ -23,40 +31,92 @@ import {
     DATASET_NAME_TIP,
     DATASPACE_TIP,
     TAG_TIP,
+    TIME_KEY_FIELD_DISPLAY,
+    TIME_KEY_FIELD_KEY,
     VISIT_DATE_TIP,
 } from './constants';
 
-export const fetchCategories = async () => {
-    // TODO: Replace this with server side call
-    return {
-        categories: [
-            { label: 'A', value: 20 },
-            { label: 'B', value: 21 },
-            { label: 'C', value: 22 },
-        ],
-    };
-};
-
-export const fetchCohorts = async () => {
-    // TODO: Replace this with server side call
-    return {
-        cohorts: [
-            { label: 'Cohort1', value: 1 },
-            { label: 'Cohort2', value: 2 },
-            { label: 'Cohort3', value: 3 },
-        ],
-    };
-};
-
-export const fetchVisitDateColumns = async () => {
-    // TODO: Keeping this action until next story in which visitDateColumns will be pulled from state change (for date fields) in the Domain Form.
-    return {
-        visitDateColumns: [
-            { label: 'Date', value: 'date' },
-            { label: 'Arrival Date', value: 'arrivalDate' },
-        ],
-    };
-};
+export function fetchCategories(): Promise<List<Option>> {
+    return new Promise((resolve, reject) => {
+        selectRows({
+            saveInSession: true,
+            schemaName: 'study',
+            sql: 'SELECT DISTINCT CategoryId.Label, CategoryId.RowId FROM DataSets',
+        })
+            .then(data => {
+                const models = fromJS(data.models[data.key]);
+                let categories = List<Option>();
+
+                data.orderedModels[data.key].forEach(modelKey => {
+                    const row = models.get(modelKey);
+                    const value = row.getIn(['Label', 'value']);
+                    const label = row.getIn(['Label', 'value']);
+
+                    categories = categories.push({ value, label });
+                });
+
+                resolve(categories);
+            })
+            .catch(response => {
+                reject(response.message);
+            });
+    });
+}
+
+export function fetchVisitDateColumns(domain: DomainDesign): List<Option> {
+    let visitDateColumns = List<Option>();
+
+    visitDateColumns = visitDateColumns.push({ value: 'date', label: 'date' });
+
+    domain.fields.map(field => {
+        if (field.rangeURI.endsWith('dateTime')) {
+            visitDateColumns = visitDateColumns.push({ value: field.name, label: field.name });
+        }
+    });
+
+    return visitDateColumns;
+}
+
+export function fetchAdditionalKeyFields(domain: DomainDesign): List<Option> {
+    let additionalKeyFields = List<Option>();
+
+    if (getServerContext().moduleContext.study.timepointType !== 'VISIT') {
```

Comment:
maybe add a comment about why this TIME_KEY_FIELD is being added here as an option